src/treegen.cpp

```diff
@@ -30,7 +30,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 namespace treegen
 {
 
-void make_tree(MMVManip &vmanip, v3s16 p0,
+void make_tree(MMVManip *vmanip, v3s16 p0,
```

Comment:
I think the syntax for const references is `const MMVManip &vmanip`. What you are doing is passing pointers by value. That is possible too, but then you need to access members via `->` and not `.`.